engine/api/workflow/workflow_run_event_test.go

```diff
@@ -0,0 +1,392 @@
+package workflow_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/ovh/cds/engine/api/application"
+	"github.com/ovh/cds/engine/api/repositoriesmanager"
+	"github.com/ovh/cds/engine/api/services"
+	"github.com/ovh/cds/engine/api/test"
+	"github.com/ovh/cds/engine/api/test/assets"
+	"github.com/ovh/cds/engine/api/workflow"
+	"github.com/ovh/cds/sdk"
+)
+
+// Test ResyncCommitStatus with a notification where all is disabled.
+// Must: no error returned, only list status is called
+func TestResyncCommitStatusNotifDisabled(t *testing.T) {
+	db, cache, end := test.SetupPG(t)
+	defer end()
+
+	ctx := context.TODO()
+
+	pkey := sdk.RandomString(10)
+	proj := assets.InsertTestProject(t, db, cache, pkey, pkey)
+	assert.NoError(t, repositoriesmanager.InsertForProject(db, proj, &sdk.ProjectVCSServer{
+		Name: "gerrit",
+		Data: map[string]string{
+			"token":  "foo",
+			"secret": "bar",
+		},
+	}))
+
+	// Create Application
+	app := sdk.Application{
+		Name:               sdk.RandomString(10),
+		ProjectID:          proj.ID,
+		RepositoryFullname: "foo/myrepo",
+		VCSServer:          "gerrit",
+		RepositoryStrategy: sdk.RepositoryStrategy{
+			ConnectionType: "ssh",
+		},
+	}
+	assert.NoError(t, application.Insert(db, cache, proj, &app))
+	assert.NoError(t, repositoriesmanager.InsertForApplication(db, &app, proj.Key))
+
+	tr := true
+	wr := &sdk.WorkflowRun{
+		WorkflowNodeRuns: map[int64][]sdk.WorkflowNodeRun{
+			1: {
+				{
+					ID:             1,
+					ApplicationID:  app.ID,
+					Status:         sdk.StatusSuccess,
+					WorkflowNodeID: 1,
+					VCSHash:        "6c3efde",
+				},
+			},
+		},
+		Workflow: sdk.Workflow{
+			WorkflowData: &sdk.WorkflowData{
+				Node: sdk.Node{
+					ID: 1,
+					Context: &sdk.NodeContext{
+						ApplicationID: app.ID,
+					},
+				},
+			},
+			Applications: map[int64]sdk.Application{
+				app.ID: app,
+			},
+			Notifications: []sdk.WorkflowNotification{
+				{
+					Settings: sdk.UserNotificationSettings{
+						Template: &sdk.UserNotificationTemplate{
+							DisableComment: &tr,
+							DisableStatus:  &tr,
+						},
+					},
+					Type: "vcs",
+				},
+			},
+		},
+	}
+
+	allSrv, err := services.LoadAll(context.TODO(), db)
+	for _, s := range allSrv {
+		if err := services.Delete(db, &s); err != nil {
+			t.Fatalf("unable to delete service: %v", err)
+		}
+	}
+	// Prepare VCS Mock
+	mockVCSSservice, _ := assets.InsertService(t, db, "TestResyncCommitStatusNotifDisabled", services.TypeVCS)
+	defer func() {
+		_ = services.Delete(db, mockVCSSservice) // nolint
+	}()
+
+	statusCall := false
+	//This is a mock for the repositories service
+	services.HTTPClient = mock(
```

Comment:
suggestion: could you mock the services call with gomock, we can make better assertions like:
https://github.com/ovh/cds/blob/master/engine/api/workflow_ascode_with_hooks_test.go#L34